flocker/control/schema/endpoints.yml

```diff
@@ -124,12 +124,14 @@ definitions:
         '$ref': 'types.json#/definitions/metadata'
       maximum_size:
         '$ref': 'types.json#/definitions/maximum_size'
-    required:
-      # Temporarily required until volume backends settle down and we know
-      # more about what it means to not have a primary manifestation.
-      - primary
     additionalProperties: false
 
+  configuration_dataset_primary:
+    description: "The configuration for a new dataset with a primary address"
+    allOf:
+      - "$ref": "#/definitions/configuration_dataset"
+      - "required": ["primary"]
```

Comment:
Please use the `- ...` list syntax everywhere rather than mixing multiple list syntaxes.